lib/usbmux/index.js

```diff
@@ -88,7 +89,16 @@ class Usbmux extends BaseServiceSocket {
    * @returns {?Object}
    */
   async readPairRecord (udid, timeout = 5000) {
-    const { tag, receivePromise } = this._receivePlistPromise(timeout);
+    const {tag, receivePromise} = this._receivePlistPromise(timeout, (data) => {
+      if (!data.payload.PairRecordData) {
+        return null;
```

Comment:
If this happens, does the _receivePlistPromise resolve to null? Won’t that cause a restructuring error?

Reply:
It returns just null as far as I remember. I haven't seen any issues there so far

Reply:
That must mean that this condition is never met, since an error will be thrown
```js
const {tag, receivePromise} = null;
Thrown:
TypeError: Cannot destructure property `tag` of 'undefined' or 'null'.
```

Reply:
But that is a different case @imurchie. Your test is failing because there is no tag field coming. In the current logic, both tag and and receivePromise objects are returned. receivePromise just resolves with a null if `data.payload.PairRecordData` doesn't exists

Reply:
Got it.